src/linking.jl

```diff
@@ -0,0 +1,75 @@
+function add_linking_constraints(m::Optimizer, λ::Dict)
```

Comment:
add an kwarg named `degree_limit::Union{Nothing,T} where {T<:Int64} = nothing,` as a control factor to control `maxdeg`. Then you can do 

````
if !isnothing(degree_limit) && degree_limit < maxdeg
        maxdeg = degree_limit
end
````

Reply:
In that case, `degree_limit` can be another user input in `solver.jl`? And it's default value to be `4` to keep the enumeration overhead low?

Reply:
Yes.